android/src/main/kotlin/com/weplenish/flutter_wifi_connect/FlutterWifiConnectPlugin.kt

```diff
@@ -274,25 +277,37 @@ class FlutterWifiConnectPlugin() : FlutterPlugin, MethodCallHandler {
   @SuppressLint("MissingPermission")
   fun connect(@NonNull wifiConfiguration: WifiConfiguration, @NonNull result: Result){
     val network = wifiManager.addNetwork(wifiConfiguration)
+    if (network == -1) {
+      result.success(false)
+      return
+    }
     wifiManager.saveConfiguration()
-    wifiManager.disconnect()
 
     val wifiChangeReceiver = object : BroadcastReceiver() {
+      var count = 0
       override fun onReceive(context: Context, intent: Intent) {
+        count++;
         val info = intent.getParcelableExtra<NetworkInfo>(WifiManager.EXTRA_NETWORK_INFO)
-        if(info != null && info.isConnected){
-          result.success(info.extraInfo == wifiConfiguration.SSID || getSSID() == wifiConfiguration.SSID)
-          context?.unregisterReceiver(this)
+        if(info != null && info.isConnected) {
+          if (info.extraInfo == wifiConfiguration.SSID || getSSID() == wifiConfiguration.SSID) {
+            result.success(true)
+            context?.unregisterReceiver(this)
+          } else if (count > 1) {
```

Comment:
This was not seen in the past since the callback was registered after calling disconnect. The changes made on `0.0.6` exposed this issue. The first `connected/connected` callback may be for the previously connected network, we ignore it if count <=1 and not successful.